includes/flattened_entity_version.js

```diff
@@ -75,65 +77,64 @@ module.exports = (params) => {
-    }
-    else {
-      var coalesceSql;
-      if (!key.pastKeyNames) {
-        coalesceSql = `DATA_struct.${key.keyName}`;
+      if (['id', 'created_at', 'updated_at'].includes(key.alias || key.keyName)) {
+        throw new Error(`${key.keyName}' is included as a field in the ${tableSchema.entityTableName}_version_${params.eventSourceName} table generated by dfe-analytics-dataform automatically, so would produce a table with more than one column with the same name. Remove this field from your dataSchema to prevent this error. Or if you're sure that you want to include the same field more than once, use an alias by setting 'alias: "alternative_name_for_${key.keyName}"' for this field in your dataSchema.`);
+      } else if (['valid_from', 'valid_to', 'event_type', 'request_uuid', 'request_path', 'request_user_id', 'request_method', 'request_user_agent', 'request_referer', 'request_query', 'response_content_type', 'response_status', 'anonymised_user_agent_and_ip', 'device_category', 'browser_name', 'browser_version', 'operating_system_name', 'operating_system_vendor', 'operating_system_version'].includes(key.alias || key.keyName)) {
+        throw new Error(`'${key.keyName}' is the same as a field name in the ${tableSchema.entityTableName}_version_${params.eventSourceName} table generated by dfe-analytics-dataform, so would produce a table with two columns with the same name. Set 'alias: "alternative_name_for_${key.keyName}"' for this field in your dataSchema to prevent this error.`);
       }
       else {
-        coalesceSql = `COALESCE(DATA_struct.${key.keyName}, DATA_struct.${key.pastKeyNames.join(', DATA_struct.')})`;
-      }
-      var fieldSql;
-      if (key.dataType == 'boolean') {
-        fieldSql = `SAFE_CAST(${coalesceSql} AS BOOL)`;
-      } else if (key.dataType == 'timestamp') {
-        fieldSql = `${data_functions.stringToTimestamp(coalesceSql)}`;
-      } else if (key.dataType == 'date') {
-        fieldSql = `${data_functions.stringToDate(coalesceSql)}`;
-      } else if (key.dataType == 'integer') {
-        fieldSql = `SAFE_CAST(${coalesceSql} AS INT64)`;
-      } else if (key.dataType == 'integer_array') {
-        fieldSql = `${data_functions.stringToIntegerArray(coalesceSql)}`;
-      } else if (key.dataType == 'float') {
-        fieldSql = `SAFE_CAST(${coalesceSql} AS FLOAT64)`;
-      } else if (key.dataType == 'json') {
-        fieldSql = `SAFE.PARSE_JSON(${coalesceSql})`;
-      } else if (key.dataType == 'string' || key.dataType == undefined) {
-        fieldSql = `${coalesceSql}`;
-      } else {
-        throw new Error(`Unrecognised dataType '${key.dataType}' for field '${key.keyName}'. dataType should be set to boolean, timestamp, date, integer, integer_array, float, json or string or not set.`);
+        var coalesceSql;
+        if (!key.pastKeyNames) {
```

Comment:
This could probably be a let instead of a var since it gets reassigned within the block